docs/blitz/design/actions/actions_list.rst

```diff
@@ -404,6 +404,53 @@ snapshot you can specify to not delete it. See `example` below.
         device: my_device
         delete_snapshot: False #optional, default is True
 
+yang_snapshot
+^^^^^^^^^^^^^
+
+The ``yang_snapshot`` action is another alternative to capture related device
+configuration in a lightweight way. It does not save all device configuration,
+instead, it scans Genie Trigger information and find out ``yang`` actions that
+will change the device configuration before the next ``yang_snapshot_restore``
+action. Note that one ``yang_snapshot`` action can be followed by one or
+multiple ``yang_snapshot_restore`` actions. When the related ``yang`` actions
+are known, ``yang_snapshot`` action is able to snapshot partial device
+configuration, which will be restored when the ``restore_config_snapshot``
+action comes.
+
+Currently the ``yang_snapshot`` action is only supported with Netconf protocol.
+More protocols will be added in the future.
+
+.. code-block:: YAML
+
+    - save_config_snapshot:
+        device: my_device
+        connection: netconf
+        protocol: netconf
+        banner: TAKE YANG SNAPSHOT 1
+
+There is one snapshot per device. A new ``yang_snapshot`` action replaces the
+snapshot created by the previous ``yang_snapshot`` action. ``banner`` is
+optional and it is just for your information.
+
+yang_snapshot_restore
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``yang_snapshot_restore`` action is used to restore a snapshot taken from
+the previous ``yang_snapshot`` action. As explained in ``yang_snapshot`` action
+above, this is not to restore a complete snapshot. The
+``yang_snapshot_restore`` action cleans up what the related ``yang`` actions
+have altered.
```

Comment:
Saying it ``is used to restore a snapshot`` is somewhat misleading because, as you described above in ``yang_snapshot``, we are not really saving device config but rather beginning a collection of yang configuration changes.  The ``yang_snapshot_restore`` is reversing yang configuration changes since the ``yang_snapshot`` began collecting.

Reply:
To be clear that we are saving device config and `yang` actions. We need both. Missing either piece of info prevents us from reversing configuration changes correctly.